File: chat-sdk/src/main/java/ru/usedesk/chat_sdk/internal/data/repository/configuration/UserInfoRepository.java
package ru.usedesk.chat_sdk.internal.data.repository.configuration;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import javax.inject.Inject;
import javax.inject.Named;
import javax.inject.Singleton;

import ru.usedesk.chat_sdk.external.entity.UsedeskChatConfiguration;
import ru.usedesk.chat_sdk.internal.data.framework.info.DataLoader;
import ru.usedesk.common_sdk.external.entity.exceptions.UsedeskDataNotFoundException;

@Singleton
public class UserInfoRepository implements IUserInfoRepository {

    private final DataLoader<UsedeskChatConfiguration> configurationDataLoader;
    private final DataLoader<String> tokenDataLoader;

    @Inject
    UserInfoRepository(@Named("configuration") DataLoader<UsedeskChatConfiguration> configurationDataLoader,
                       @Named("token") DataLoader<String> tokenDataLoader) {
        this.configurationDataLoader = configurationDataLoader;
        this.tokenDataLoader = tokenDataLoader;
    }

    @Override
    @NonNull
    public String getToken() throws UsedeskDataNotFoundException {
        return tokenDataLoader.getData();
    }

    @Override
    public void setToken(@Nullable String token) {
        tokenDataLoader.setData(token);
    }

    @Override
    @NonNull
    public UsedeskChatConfiguration getConfiguration() throws UsedeskDataNotFoundException {
        return configurationDataLoader.getData();
    }

    @Override
    public void setConfiguration(@Nullable UsedeskChatConfiguration configuration) {
        configurationDataLoader.setData(configuration);
    }
}

File: chat-sdk/src/main/java/ru/usedesk/chat_sdk/internal/data/repository/configuration/UserInfoRepository.kt
package ru.usedesk.chat_sdk.internal.data.repository.configuration

import ru.usedesk.chat_sdk.external.entity.UsedeskChatConfiguration
import ru.usedesk.chat_sdk.internal.data.framework.configuration.IConfigurationLoader
import ru.usedesk.chat_sdk.internal.data.framework.token.ITokenLoader
import ru.usedesk.common_sdk.external.entity.exceptions.UsedeskDataNotFoundException
import toothpick.InjectConstructor

@InjectConstructor
class UserInfoRepository(
        private val configurationDataLoader: IConfigurationLoader,
        private val tokenDataLoader: ITokenLoader
) : IUserInfoRepository {

    @Throws(UsedeskDataNotFoundException::class)
    override fun getToken(): String {
        return tokenDataLoader.getData()
    }

    override fun setToken(token: String?) {
        if (token == null) {
            tokenDataLoader.clearData()
        } else {
            tokenDataLoader.setData(token)
        }
    }

    @Throws(UsedeskDataNotFoundException::class)
    override fun getConfiguration(): UsedeskChatConfiguration {
        return configurationDataLoader.getData()
    }

    override fun setConfiguration(configuration: UsedeskChatConfiguration?) {
        if (configuration == null) {
            configurationDataLoader.clearData()
        } else {
            configurationDataLoader.setData(configuration)
        }
    }
}